### PR TITLE
feat: switch to using beehiiv for signups

### LINKED ISSUE #11: Figure out Beehiiv API
Link: https://developers.beehiiv.com/docs/v2/1f82a0eaf9b68-create

Worried that if I integrate this while I'm on the free trial, it won't work after that's up 🤔 Not sure though so may need to reach out to support and ask.

### PR DESCRIPTION
**Pull request by jsjoeio**
## Description

This moves over the waitlist sign up to Beehiiv rather than Reform.

## Changes
- [x] add `BEEHIIV_API_TOKEN` to `.env` and Vercel
- [x] update `src/pages/getWaitlistTotal.json.ts`
- [x] update `src/components/CTA.tsx` to post to Beehiiv
- [x] redirect to thank you 
- [x] show loading spinner when submitting
- [x] fix React hydration error

```js
act-dom.development.js:86 Warning: Did not expect server HTML to contain a <div> in <form>.
    at form
    at div
    at div
    at CTA (http://localhost:3000/src/components/CTA.tsx:21:45)
p
```

Fixes #11

### CODE REVIEW

File: src/components/CTA.tsx [hunk 1]
```diff
@@ -1,22 +1,36 @@
...
+      const data = (await res.json()) as ResponseData;
 
-    // Submit the form
-    event.target.submit();
-  };
-  type ResponseData = {
-    totalCount: number;
+      if (data.success) {
+        // help
+        await redirect("/thank-you");
```
Comment (jsjoeio) [inline src/components/CTA.tsx:25]: @coding-in-public do you know how to redirect to another route in Astro in a React component? I have spent 45mins trying lol and cannot figure it out. 

what i've tried:
- passing Astro.redirect as a prop to my React component
- calling redirect in my post API endpoint
- window.location.href
- windows.location.push

Comment (coding-in-public) [reply src/components/CTA.tsx]: Hmm…@jsjoeio you should be able to just replace the URL, no? like this…

 location.replace("/thank-you");

Comment (jsjoeio) [reply src/components/CTA.tsx]: Ah thank you! Got it working :)

Comment (jsjoeio) [reply src/components/CTA.tsx]: This was the issue FWIW: https://twitter.com/jsjoeio/status/1651634164788248576?s=20

**Pull request merged**